### ISSUE TITLE
Get final OCP core entitlement documented

### ISSUE THREAD

**Issue by stewartshea** [BCDevOps/OpenShift4-RollOut#114]
For the architecture documents we need to know the total core count so that we can divy them up between each cluster .

**Comment by Joe-Craig** [comment]
![RedHat-OpenShift-Licenses](https://user-images.githubusercontent.com/45772546/68977446-18800080-07ad-11ea-8dbd-f8c58f15e63e.JPG)

**Comment by stewartshea** [comment]
@Joe-Craig What about the ESXi Lab Environment? They were fully entitled sockets... did those get converted in the purchase? 

Also, can you confirm that the above are the actual entitlement numbers, which are used per vCPU pair. The math being that that we have 416 entitlements which allows for 832 vCPU cores. Thanks!

**Comment by Joe-Craig** [comment]
@stewartshea the 208 includes the licenses for the ESXi cluster, so yes they were converted as part of the deal. So we can either license the underlying virtual host and build as many clusters or license the machines running on the VMs.
![image.png](https://images.zenhubusercontent.com/5dcf1d4541ec08000170595e/637389c4-f18e-43c9-afd5-c8c240833059)

**Comment by Joe-Craig** [comment]
![image.png](https://images.zenhubusercontent.com/5dcf1d4541ec08000170595e/8d76c0f7-789b-4b0a-b3d8-95ab4e8d904b)

**Comment by stewartshea** [comment]
That's a sneaky move; re: hyperthreading. ugh.

**Comment by StevenBarre** [comment]
@stewartshea isn't that a good thing? All the ESXi hosts have HT enabled. So a 4 cpu worker VM only uses 2 cores of licensing, or one license unit (since each unit is for 2 cores).

**Comment by stewartshea** [comment]
@sbarre-esit I think I might be reading his responses the other way around; to be honest the examples and responses above aren't very clear. For example, if you had 132 cores, you wouldn't license 132x core price, your would license half that amount since every entitlement covers 2 cores. And then the example jumps to hyperthreading and 20 or 40 cores. It's still unclear to me, but maybe I'm just reading it all wrong. I'll go digging into the latest licensing docs to see if I can find clarity.

**Comment by StevenBarre** [comment]
@stewartshea So looking at the 96 seats for the 12 new nodes we can do some maths. The new nodes are dual socket 8 core processors with hyper threading = 16 real cores and 32 cores visible to OS. 

96/12=8 So each server is using 8 seats or 4 virtual cores to a seat.

**Comment by stewartshea** [comment]
Thanks @sbarre-esit I appreciate the comment and the example. I was hoping not to rely on existing assumptions (ie. assumptions that what was purchased for the existing nodes was indeed the correct amount of entitlements). 

That said, your math is correct. Validated  this up from the latest sizing and licensing guide; 

Virtual machines use virtualized CPUs.

For hyperthreaded systems
- To determine whether a particular system supports hyperthreading, visit https://access.redhat.com/solutions/7714 
- You can see two vCPU per underlying physical core
- Red Hat calculates cores with a ratio of 2 core = 4 vCPUs
- In other words, a 2-core subscription covers 4 virtual CPUs in a VM
-      Example: An 8 vCPU VM has 4 effective “cores”, and would need two (2) 2 core subscriptions.
- For non-hyperthreaded systems

You can see one vCPU per underlying physical core
- Red Hat calculates cores with a ratio of 2 cores = 2 vCPUs
- In other words, a 2-core of a subscription covers 2 virtual CPUs in a VM on a non-hyperthreaded system
-      Example: An 4 vCPU VM has 4 effective “cores”, and would need two (2) 2 core subscriptions.
